cpp/csp/cppnodes/statsimpl.h

```diff
@@ -12,6 +12,10 @@
 #ifdef __linux__
 #include <ext/pb_ds/assoc_container.hpp>
 #include <ext/pb_ds/tree_policy.hpp>
+#else
+#include <boost/multi_index_container.hpp>
```

Comment:
Hey @mrchhaya , thanks for the contribution!

I tested your changes on Linux using a simple benchmarking script below and the results show that the `boost::multi_index` data structure is actually even faster than the `gnu::pbds` tree! Thus, we should just remove the ost implementation and cut over fully to your approach. This will also make the code less complicated as we can remove the `#ifdef` statements that are all over the place. 

If you're curious, here is the script I used for benchmarking:

```python
import csp
import numpy as np
from datetime import datetime, timedelta
import time

st = datetime(2020,1,1)
N = 10_000
ARRAY_SIZE = 100
TEST_TIMES = [st+timedelta(seconds=i) for i in range(N)]
RANDOM_VALUES = [np.random.normal(size=(ARRAY_SIZE,)) for i in range(N)] # 100 element np array
DATA = list(zip(TEST_TIMES, RANDOM_VALUES))
INTERVAL = 1000
NUM_SAMPLES = 100

def g_qtl():
    data = csp.curve(typ=np.ndarray, data=DATA)
    median = csp.stats.median(data, interval=INTERVAL)
    csp.add_graph_output('final_median', median, tick_count=1)

def g_rank():
    data = csp.curve(typ=np.ndarray, data=DATA)
    rank = csp.stats.rank(data, interval=INTERVAL)
    csp.add_graph_output('final_rank', rank, tick_count=1)

if __name__ == '__main__':
    qtl_times, rank_times = list(), list()
    for i in range(NUM_SAMPLES):
        start = time.time()
        csp.run(g_qtl, starttime=st, endtime=timedelta(seconds=N))
        post_qtl = time.time()
        csp.run(g_rank, starttime=st, endtime=timedelta(seconds=N))
        post_rank = time.time()

        qtl_times.append(post_qtl-start)
        rank_times.append(post_rank-post_qtl)
        print(i)

    avg_med = sum(qtl_times)/NUM_SAMPLES
    avg_rank = sum(rank_times)/NUM_SAMPLES
    print(f'Average time in {NUM_SAMPLES} tests for median with {N=}, {ARRAY_SIZE=}, {INTERVAL=}: {round(avg_med, 2)} s')
    print(f'Average time in {NUM_SAMPLES} tests for rank with {N=}, {ARRAY_SIZE=}, {INTERVAL=}: {round(avg_rank, 2)} s')
```

and here are the results I got:

boost::multi_index

```
Average time in 100 tests for median with N=10000, ARRAY_SIZE=100, INTERVAL=500: 0.6 s
Average time in 100 tests for rank with N=10000, ARRAY_SIZE=100, INTERVAL=500: 0.57 s

Average time in 100 tests for median with N=10000, ARRAY_SIZE=100, INTERVAL=1000: 0.68 s
Average time in 100 tests for rank with N=10000, ARRAY_SIZE=100, INTERVAL=1000: 0.64 s
```

gnu::pbds::tree

```
Average time in 100 tests for median with N=10000, ARRAY_SIZE=100, INTERVAL=500: 0.67 s
Average time in 100 tests for rank with N=10000, ARRAY_SIZE=100, INTERVAL=500: 0.63 s

Average time in 100 tests for median with N=10000, ARRAY_SIZE=100, INTERVAL=1000: 0.73 s
Average time in 100 tests for rank with N=10000, ARRAY_SIZE=100, INTERVAL=1000: 0.68 s
```

Reply:
@AdamGlustein we should canonicalize your benchmark script, perhaps start on a `csp/benchmarks` folder in python. I'd like to get [airspeed-velocity](https://asv.readthedocs.io/en/v0.6.1/) setup at some point.

Reply:
ref: https://github.com/Point72/csp/issues/38

Reply:
starting here: https://github.com/Point72/csp/pull/332

Reply:
pretty picture of the benchmark results:

<img width="1473" alt="Screenshot 2024-07-13 at 19 00 24" src="https://github.com/user-attachments/assets/8ae34964-3225-4a71-8d03-8da772160dde">